linera-core/src/chain_worker/state/mod.rs

```diff
@@ -309,26 +309,78 @@ where
     }
 
     /// Returns an error if the block requires a blob we don't have, or if unrelated blobs were provided.
-    async fn check_no_missing_blobs(
+    async fn check_no_missing_confirmed_blobs(
```

Comment:
Same idea that I described below applies here. This is just a confirmed block that may or may not be publishing blobs, it could just be using them.